parser.hpp

```diff
@@ -97,21 +117,32 @@ CONSTEXPR CandyState decode_candy_state(char up_char)
 
 constexpr int candy_size = 3;
 constexpr int candy_type_offset = 1;
+constexpr int row_padding = 5;    // space, pipe, pipe, space, newline
+constexpr int noncandy_rows = 3;  // title, border, border
 
-constexpr auto parse_board_column_count(const constexpr_string_view& str)
+constexpr auto parse_board_parameters(auto&& str)
```

Comment:
Since this is a string_view, I actually think that here the best would be to take by value

Reply:
Done! Now send by value